src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Imaging/ImageMonikerDebuggerDisplay.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.ProjectSystem;
+
+using ImageMoniker = Microsoft.VisualStudio.Imaging.Interop.ImageMoniker;
+
+[assembly: DebuggerDisplay("{Microsoft.VisualStudio.ProjectSystem.Imaging.ImageMonikerDebuggerDisplay.FromImageMoniker(this)}", Target = typeof(ImageMoniker))]
+[assembly: DebuggerDisplay("{Microsoft.VisualStudio.ProjectSystem.Imaging.ImageMonikerDebuggerDisplay.FromProjectImageMoniker(this)}", Target = typeof(ProjectImageMoniker))]
+
+namespace Microsoft.VisualStudio.ProjectSystem.Imaging
+{
+    /// <summary>
+    ///     Provides a friendly display for <see cref="ProjectImageMoniker"/> and <see cref="ImageMoniker"/> instances
+    ///     based on the well known values from <see cref="KnownImageIds"/>.
+    /// </summary>
+    internal static class ImageMonikerDebuggerDisplay
+    {
+        private static readonly Dictionary<(Guid, int), string> s_displayNames = CalculateDisplayNames();
+
+        internal static string FromImageMoniker(ImageMoniker moniker) => DebugDisplay(moniker.Guid, moniker.Id);
```

Comment:
This should move to the package containing `Microsoft.VisualStudio.Imaging` and be made public for use in test code and for other types such as `ProjectImageMoniker`.

Reply:
I wonder if the code generation in `ManifestToCode` could be extended to help here.